stravalib/model.py

```diff
@@ -818,460 +725,61 @@ class SegmentEffort(BaseEffort):
-    segment_efforts = EntityCollection(
-        SegmentEffort, (DETAILED,)
-    )  #: :class:`list` of :class:`stravalib.model.SegmentEffort` efforts for activity.
-    splits_metric = EntityCollection(
-        Split, (DETAILED,)
-    )  #: :class:`list` of metric :class:`stravalib.model.Split` summaries (running activities only)
-    splits_standard = EntityCollection(
-        Split, (DETAILED,)
-    )  #: :class:`list` of standard/imperial :class:`stravalib.model.Split` summaries (running activities only)
-
-    average_watts = Attribute(
-        float, (SUMMARY, DETAILED)
-    )  #: (undocumented) Average power during activity
-    weighted_average_watts = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  # rides with power meter data only similar to xPower or Normalized Power
-    max_watts = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  #: rides with power meter data only
-
-    suffer_score = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  #: a measure of heartrate intensity, available on premium users' activities only
-    has_heartrate = Attribute(
-        bool, (SUMMARY, DETAILED)
-    )  #: true if recorded with heartrate
-    average_heartrate = Attribute(
-        float, (SUMMARY, DETAILED)
-    )  #: only if recorded with heartrate average over moving portion
-    max_heartrate = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  #: (undocumented) Max HR during activity
-    average_cadence = Attribute(
-        float, (SUMMARY, DETAILED)
-    )  #: (undocumented) Average cadence during activity
-    kilojoules = Attribute(
-        float, (SUMMARY, DETAILED)
-    )  #: (undocumented) Kilojoules of energy used during activity
-    average_temp = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  #: (undocumented) Average temperature (when available from device) during activity.
-
-    device_name = Attribute(
-        str, (SUMMARY, DETAILED)
-    )  #: the name of the device used to record the activity.
-    embed_token = Attribute(
-        str, (DETAILED,)
-    )  #: the token used to embed a Strava activity in the form www.strava.com/activities/[activity_id]/embed/[embed_token]. Only included if requesting athlete is activity owner.
-    calories = Attribute(
-        float, (DETAILED,)
-    )  #: Calculation of how many calories burned on activity
-    description = Attribute(str, (DETAILED,))  #: Description of activity.
-    workout_type = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  #: The activity's workout type
-
-    photos = EntityAttribute(
-        ActivityPhotoMeta, (DETAILED,)
-    )  #: A new photo metadata structure.
-    instagram_primary_photo = Attribute(
-        str, (DETAILED,)
-    )  #: (undocumented) Appears to be the ref to first associated instagram photo
-
-    partner_logo_url = Attribute(str, (DETAILED,))  #: (undocumented)
-    partner_brand_tag = Attribute(str, (DETAILED,))  #: (undocumented)
+    _field_conversions = {
```

Comment:
ah... ok excited to try this locally. i think this might be why i was getting some odd data returned last time (in the last pr!)